modules/Bio/EnsEMBL/Compara/GeneTreeNode.pm

```diff
@@ -312,7 +312,9 @@ sub is_leaf {
     if ( $child_count == 0 ) {
         return 1;
     } elsif ( $child_count == 1 && $self->tree->tree_type eq 'supertree' ) {
-        return 1;
+        my $child = $self->children->[0];
+        return 1 if $child->node_id == $child->root->node_id;
+        return 0;
```

Comment:
This is just out of curiosity, could you do `return ($child->node_id == $child->root->node_id)`?

Reply:
Yeah, I like that
```suggestion
        return ($child->node_id == $child->root->node_id);
```